packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1476,13 +1475,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             const electedSummarizerContent = JSON.stringify(this.summarizerClientElection?.serialize());
             addBlobToSummary(summaryTree, electedSummarizerBlobName, electedSummarizerContent);
         }
-        const snapshot = this.blobManager.snapshot();
 
+        const summary = this.blobManager.summarize() as ISummaryTreeWithStats;
```

Comment:
I'm not sure I follow why you cast it to ISummaryTreeWithStats. This seems wrong, given that actual result returned is ISummarizeResult.

Reply:
I'm glad you brought this up - perhaps I took the wording from #8997 too literally. I don't see a way to preserve the current behavior of omitting the empty tree where we return an ISummarizeResult in summarize() and are able to discern whether the tree is empty without casting to an ISummaryTreeWithStats (unless I'm missing something, which is entirely possible). It seems like it would be preferable to have summarize() return an ISummaryTreeWithStats instead. Thoughts?

Reply:
I'm not sure why I don't get a compile error when returning an ISummarizeResult rather than an ISummaryTreeWithStats. From what I can tell (I'm not very familiar with TypeScript yet), an ISummarizeResult returns a SummaryTree rather than an ISummaryTree, but a SummaryTree is exported as `ISummaryTree | ISummaryHandle`. Because of this, this code is also valid:
```
if (summary.summary.type === SummaryType.Tree) {
    if (Object.keys(summary.summary.tree).length > 0) {
        addTreeToSummary(summaryTree, blobsTreeName, summary);
    }
}
```

Regardless of this, I'm changing the return type of summarize() to an ISummaryTreeWithStats to avoid the cast. It does seem strange that the compiler is ok with this return type though.